ostd/src/arch/x86/trap/gdt.rs

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MPL-2.0 OR MIT
+
+//! Configure Global Descriptor Table (GDT).
+//!
+//! The most contents of this file come from https://github.com/rcore-os/trapframe-rs.
```

Comment:
First of all, do not include license or source origin information as Rust doc. Just make them regular comments. Second, it does not say the original code is licensed under MIT. Third, it does not clear which parts are changed by us, which is protected by our copyright license (MPL).

So I suggest changing the file to the following form.

```rust
// The original source code is from [trapframe-rs](https://github.com/rcore-os/trapframe-rs),
// which licensed under the following license.
//
// SPDX-License-Identifier: MIT
//
// Copyright (c) 2020 - 2024 The authors of trapframe-rs 
//
// We make the following new changes:
// * Change 1
// * Change 2
//
// These changes are released under the following license:
//
// SPDX-License-Identifier: MPL-2.0
// 
```